frontend/src/scenes/surveys/Survey.tsx

```diff
@@ -179,10 +179,20 @@ export function SurveyDisplaySummary({
                     <span>Feature flag enabled for:</span>{' '}
                     {id !== 'new' ? (
                         survey.linked_flag?.id ? (
-                            <Link to={urls.featureFlag(survey.linked_flag?.id)}>{survey.linked_flag?.key}</Link>
+                            <>
+                                <Link to={urls.featureFlag(survey.linked_flag?.id)}>{survey.linked_flag?.key}</Link>
+                                {survey.conditions?.linkedFlagVariant && (
+                                    <LemonTag>variant: {survey.conditions.linkedFlagVariant}</LemonTag>
+                                )}
+                            </>
                         ) : null
                     ) : (
-                        <FlagSelector value={survey.linked_flag_id} readOnly={true} onChange={() => {}} />
+                        <>
+                            <FlagSelector value={survey.linked_flag_id} readOnly={true} onChange={() => {}} />
+                            {survey.conditions?.linkedFlagVariant && (
+                                <LemonTag>variant: {survey.conditions.linkedFlagVariant}</LemonTag>
+                            )}
```

Comment:
left a question about this https://posthog.slack.com/archives/C07Q0K8H0CB/p1754399504470269

Reply:
i think the reason is the same as https://github.com/PostHog/posthog/pull/36200/files#r2254270985
the check is either `survey.linked_flag_id` or `survey.linked_flag?.id`
`survey.linked_flag` only has a value after the survey creation.